test/cypress/support/commands.js

```diff
@@ -28,7 +28,10 @@ Cypress.Commands.add('giLogin', (userName, password = '123456789') => {
   cy.getByDT('password').clear().type(password)
 
   cy.getByDT('loginSubmit').click()
+})
 
+Cypress.Commands.add('giLoginWithGroup', (userName, password = '123456789') => {
+  cy.giLogin(userName)
```

Comment:
In cypress we can create conditions in our tests to avoid "similar but not equal steps" like giLogin and giLoginWithGroup. In this case, the last step can be to verify the message, something like this:

```js
Cypress.Commands.add('giLogin', (userName, password = '123456789') => {
  // login...
  
  const hasNoGroup = cy.get('#app').find('[data-test="welcomeHomeLoggedIn"]') // maybe welcomeWithoutGroup would be better

  if (hasNoGroup) {
      cy.getByDT('welcomeHomeLoggedIn').should('contain', 'Let’s get this party started')
  } else {
     // the code it was before to get the displayName and verify it.
  }
})
```

The same technique can be used to logout as well. Find an element. based on its existence, do A or B assertion.